### PR TITLE
[wallet-ext] build new "Protect Account" UI screen and add some form primitives

### PR DESCRIPTION
**Pull request by williamrobertson13**
## Description 

This PR adds the new "Protect Account" screen used when clicking "Create a new account" button and as the second step of the "Import Passphrase" and "Import Private Key" flows. As part of this, I attempted to create some new form primitives to make building forms more manageable (all of the existing components aren't quite up to spec, use Formik, and aren't fully accessible). I ended up following an approach similar to https://www.brendonovich.dev/blog/the-ultimate-form-abstraction which gave some creds to @Jordan-Mysten 😆 

As a rough outline, we have generic, non-form-library-specific input controls like `TextArea`, `Input`, `PasswordInput`, `Checkbox` which are used to create `react-hook-form` specific controls such as `TextField`, `TextAreaField`, `CheckboxField`, and so forth. We also have some helper components like `Form` and `FormField` to help abstract away some specific form details such as error states when using react-hook-form. I considered using the Radix form primitives, but I didn't really see the immediate value 🤷🏼 

Additional note #1: Some of these pages are used in different flows and have different submission logic depending on the usage context. I think I might need to brainstorm on the best way to handle that and tackle it in a follow-up PR

Additional note #2: the auto-lock input is still a WIP on the design side, so I have a TODO to add that once it's ready.

Checkbox in Figma - https://www.figma.com/file/T06obgYVOUD2JDGXM8QEDX?node-id=341%3A378&main-component=1&fuid=1209977329759347633

Input in Figma - https://www.figma.com/file/T06obgYVOUD2JDGXM8QEDX/01-Components-%3A-Shared?node-id=19%3A312&mode=dev

image

## Test Plan 
- Manual testing (error states, successful submission, accessibility, focus/disabled/hover states, etc.)
- CI

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/pages/accounts/ProtectAccountPage.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+				Wallet Setup
+			</Text>
+			<div className="text-center mt-2.5">
+				<Heading variant="heading1" color="gray-90" as="h1" weight="bold">
+					Protect Account with a Password Lock
+				</Heading>
+			</div>
+			<div className="mt-6 w-full grow">
+				<ProtectAccountForm
+					cancelButtonText="Back"
```
Comment (williamrobertson13) [inline apps/wallet/src/ui/app/pages/accounts/ProtectAccountPage.tsx:21]: This form will be re-used on a different flow hence the customizable button text

File: apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		return (
+			<div className="flex w-full relative items-center">
+				<Input
+					type={passwordShown ? 'text' : 'password'}
+					placeholder="Password"
+					ref={forwardedRef}
+					{...props}
+				/>
+				<ButtonOrLink
+					className="appearance-none bg-transparent border-none cursor-pointer absolute right-3 text-gray-60 peer-focus:text-steel"
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx:27]: looks like we could use a `flex items-center` here: 

image

File: apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		const [passwordShown, setPasswordShown] = useState(false);
+		const IconComponent = passwordShown ? EyeOpen16 : EyeClose16;
+
+		return (
+			<div className="flex w-full relative items-center">
+				<Input
+					type={passwordShown ? 'text' : 'password'}
+					placeholder="Password"
+					ref={forwardedRef}
+					{...props}
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx:24]: I think this is breaking the show/hide password button as `type="password"` is being passed through from `ProtectAccountForm`

File: apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}: ProtectAccountFormProps) {
+	const form = useForm({
+		mode: 'all',
+		defaultValues: {
+			password: '',
+			confirmedPassword: '',
+			acceptedTos: false,
+			enabledAutolock: true,
+		},
+		resolver: yupResolver(formSchema),
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx:43]: could be nice to attempt to achieve the equivalent of `useZodForm` with `Yup` (or ditch Yup at some point soon)

**Review by mamos-mysten — APPROVED** [review]
nice and clean 🧘🏻‍♂️

File: apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx (hunk 3 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx]: ah good catch, I fixed this by spreading the props first then overriding the type

File: apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx (hunk 2 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx]: good 👁️ , thanks

File: apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx (hunk 4 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx:43]: yeah ditching `Yup` would be nice, I just didn't want to spend the time re-writing all of the validation logic haha

File: apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import * as Yup from 'yup';
+import { CheckboxField } from '../../shared/forms/CheckboxField';
+import { Form } from '../../shared/forms/Form';
+import { TextField } from '../../shared/forms/TextField';
+import ExternalLink from '../external-link';
+import { Button } from '_app/shared/ButtonUI';
+import { ToS_LINK } from '_src/shared/constants';
+
+const formSchema = Yup.object({
+	password: Yup.string().required('Required'),
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx:16]: Sorry I missed this, can we use zod for new things instead?

Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/accounts/ProtectAccountForm.tsx:16]: sure thing!

**Pull request merged**